packages/backend-common/src/database/DatabaseManager.ts

```diff
@@ -89,7 +89,30 @@ export class DatabaseManager {
     private readonly prefix: string = 'backstage_plugin_',
     private readonly options?: DatabaseManagerOptions,
     private readonly databaseCache: Map<string, Promise<Knex>> = new Map(),
-  ) {}
+  ) {
+    if (this.config.subscribe) {
+      console.log('-----==== Subscribe() IS supported on config in DatabaseManager ===----');
+      this.config.subscribe(() => {
+        console.log('-----==== Config change detected in DatabaseManager ===----');
+        console.log(JSON.stringify(this.config));
+        const databaseConfig = this.config.getOptionalConfig('backend.database');
+        if (databaseConfig) {
+          console.log(`DatabaseConfig: ${databaseConfig}`);
+        } else {
+          console.log('Couldn\'t find backend.database');
+          console.log(`Has backend: ${this.config.has('backend')}`);
+          console.log(`Keys: ${config.keys()}`);
+          console.log(`parent: ${JSON.stringify(config.getOptionalConfig('parent'))}`);
+          console.log(`backend: ${JSON.stringify(config.getOptionalConfig('backend'))}`);
+        }
+        console.log('Invalidating the database cache...');
+        this.databaseCache.clear();
```

Comment:
To add to the above Q, I'm wondering what the behavior would be if we provide an `expirationChecker` that always returns `true`.

Reply:
> Do we need to tell knex that config has changed? Would it be enough to simply read the latest config whenever we want to create a new database connection? Reading config from the ConfigService is very fast, more or less property-access fast. With that I feel we would be able to avoid quite a lot of complicated coupling?

I don't think you can "just" read the latest config. In token-based access architecture, you will have to make a request to acquire a new token. Together with this token, you will get an expiration timestamp. Simple comparison current timestamp will not take long to mark the connection as expired.

There is a similar issue and PR with more accomplished implementation of the approach proposed here https://github.com/backstage/backstage/pull/18507/files

Reply:
That's not what we're doing now though right? At the moment we are just reacting to changes in config.

Reply:
Well, yeah. Token-based access is out of the scope of this PR. Sorry :)

Reply:
So is it alright to flip around the approach in this PR? So that rather than subscribing and pushing config changes we have the knex client read the latest config every time it needs to create a new connection? We just gotta make sure that the `expirationChecker` behaves well with that approach. As in if we use a naive approach were it always returns `true`, we'd for example need to make sure that doesn't cause existing connections to be torn down. It could also be that we actually want the `expirationChecker` to read config itself, to check if it has changed, if that ends up being the best approach we'd also need to make sure it isn't called too often, i.e. on every statement.